### DISCUSSION TITLE
Custom tts-dictionary?

### DISCUSSION THREAD

**Discussion by Glazkor** [neonbjb/tortoise-tts#140] [Ideas]
Any way for tortoise-tts to convert the samples into an tts-dictionary / corresponding file, mainly compatible for things like pyaudio and the likes to create a bridge between tortoise-tts, speech recognition and tts-models would be helpful.

**Comment by neonbjb** [comment]
I'm not entirely sure what you mean by "tts-dictionary" - googling the term seems to indicate it's conventionally used to refer to lookup tables to map abbreviations to words (e.g. "OMG"->"Oh my god"). From the context of your statement that doesn't seem to be what you mean, though. Can you clarify further?

**Comment by Glazkor** [comment]
Sorry if I was a bit misleading, I explain my point.
TTS-software nowadays, as I am aware of, has two different ways to create Audio. The first approach is to train a voice or create a voice through a CMU pronouncing dictionary, make it into a file and have a finished tts-voice for reading (how Microsoft's TTS works or how pyaudio and pyttsx3 work with SAPI).
The second approach is the one tortoise-tts uses, which is by creating a temporary voice out of clip samples and by using an autoregressive and a diffusion decoder. This makes it slow and not suited for creating long sentences, to read things aloud or do continuous projects with the same voice, as the same "speaker" needs to be synthesized anew.
What has never been done before is creating a file from the clip samples of tortoise-tts, which then can be integrated into, for example pyttsx3 or read executed by tortoise-tts to read faster or even at real-time.
I don't know if this can be executed or if I am right with this observation as I do not claim to know more than the creator does, but it would be an option that solves the problems and faults tortoise has at it's core.

Oh and thank you for the project, I appreciate it very much.
